frontend/packages/kubevirt-plugin/src/selectors/vm-template/advanced.ts

```diff
@@ -49,6 +56,14 @@ export const getTemplatesLabelValues = (templates: TemplateKind[], label: string
 export const getTemplateFlavors = (templates: TemplateKind[]) =>
   getTemplatesLabelValues(templates, TEMPLATE_FLAVOR_LABEL);
 
+export const getTemplateHostname = (template: TemplateKind) => {
+  const vm = selectVM(template);
+  const cloudInitVolume = getVolumes(vm).find((vol) => !!getVolumeCloudInitUserData(vol));
+  const data = getVolumeCloudInitUserData(cloudInitVolume);
+  const cloudInitHelper = new CloudInitDataHelper({ userData: data });
```

Comment:
you should pass a `cloudInitNoCloud` here. There is no guarantee the data is stored in userData.
Either use `VolumeWrapper.initialize(volume).getCloudInitNoCloud()` or create a selector for that.